src/main/java/com/rocana/lucene/codec/v1/RocanaPerFieldPostingsFormat.java

```diff
@@ -274,6 +304,30 @@ public FieldsReader(final SegmentReadState readState) throws IOException {
       this.segment = readState.segmentInfo.name;
     }
 
+    /**
+     * Return our fork of Lucene's postings format.
+     *
+     * The original Lucene code did an SPI (Service Provider Interface) lookup to
+     * get the postings format. The problem is we forked the postings format, but
+     * didn't register it via SPI -- for the reason see:
+     * {@link RocanaSearchCodecV1#getActualPostingsFormat()}
+     *
+     * This meant our custom codec was using our forked postings format, but here
+     * the SPI lookup returned Lucene's original postings format. So we're mixing
+     * postings formats.
+     *
+     * In this method we'll catch those lookup calls for Lucene's postings format
+     * ("Lucene50") and instead return Rocana's fork.
+     */
+    private PostingsFormat lookupPostingsFormat(String postingsFormatShortName) {
+      if (postingsFormatShortName.equals("Lucene50")) {
+        RocanaSearchCodecV1 codec = (RocanaSearchCodecV1) Codec.forName(RocanaSearchCodecV1.NAME);
```

Comment:
There is a loading cost with no benefit to calling `Codec.forName` and then casting to a concrete type.  You should either just instantiate the concrete type directly (faster) or create an interface `RocanaSearchCodec` with the `getActualPostingsFormat` (extensible).